hack/jenkins/build.sh

```diff
@@ -54,9 +57,25 @@ git clean -fdx
 # Build
 go run ./hack/e2e.go -v --build
 
-[[ ${KUBE_SKIP_PUSH_GCS:-} =~ ^[yY]$ ]] || {
-    # Push to GCS
-    ./build/push-ci-build.sh
-}
+# Push to GCS?
+if [[ ${KUBE_SKIP_PUSH_GCS:-} =~ ^[yY]$ ]]; then
+  echo "Not pushed to GCS..."
+elif $RELEASE_INFRA_PUSH; then
```

Comment:
Another nice option in bash is to use 0 or 1 for your bool values and take advantage of the double paren eval.  Both are certainly more Go or Python like than the overly decorated expression above (== true):

So both read nicely:
```
RELEASE_INFRA_PUSH=1
if ((RELEASE_INFRA_PUSH)); then
```
Or:
```
RELEASE_INFRA_PUSH=true
if $RELEASE_INFRA_PUSH; then
```